### PR TITLE
feat: encrypted telegram handle on-chain

### PR DESCRIPTION
**Pull request by davirds**

### CODE REVIEW

File: app/src/ui/components/offers/CreateOffer.vue [hunk 1]
```diff
@@ -65,17 +72,24 @@ document.documentElement.style.setProperty('--vh', `${vh}px`)
...
+  const telegramHandle = removeTelegramURLPrefix(ownerContact.value)
+  console.log('owner_contact: ', telegramHandle)
+  // Encrypt contact to save on the profile when an offer is created
+  const owner_contact_pk = secrets.value.publicKey
+  const owner_contact = await encryptData(owner_contact_pk, telegramHandle)
+
   const postOffer: PostOffer = {
-    owner_contact: removeTelegramURLPrefix(ownerContact.value),
+    owner_contact,
+    owner_encrypt_key: owner_contact_pk,
```
Comment (samb-local) [inline app/src/ui/components/offers/CreateOffer.vue:84]: I think that `owner_encryption_key` or `owner_contact_pubkey` would be better names for these values, also I would rename one of these two to match the other so they're the same name.

Comment (davirds) [reply app/src/ui/components/offers/CreateOffer.vue:84]: Yeah ... still in development, I will change to better names

File: contracts/contracts/offer/src/contract.rs [hunk 2]
```diff
@@ -162,13 +170,25 @@ pub fn update_offer(
 ) -> Result<Response, ContractError> {
     assert_min_g_max(msg.min_amount, msg.max_amount)?;
 
+    let hub_config = get_hub_config(deps.as_ref());
     let mut offer_model = OfferModel::may_load(deps.storage, &msg.id);
 
-    assert_ownership(info.sender, offer_model.offer.owner.clone())?;
+    assert_ownership(info.sender.clone(), offer_model.offer.owner.clone())?;
+
+    let mut send_msgs: Vec<SubMsg> = Vec::new();
```
Comment (samb-local) [inline contracts/contracts/offer/src/contract.rs:178]: I think that this should be better named as `sub_msgs` or something similar. 

imo, this name (`send_msgs`) would better represent a vec of `Bank::Send` msgs, like we have in `release_escrow`, etc.

File: app/src/ui/components/offers/CreateOffer.vue [hunk 3]
```diff
@@ -65,17 +72,24 @@ document.documentElement.style.setProperty('--vh', `${vh}px`)
 function calculateMarginRate() {
   rate.value = convertMarginRateToOfferRate(margin.value, marginOffsetUnmasked.value)
 }
-function createOffer() {
+async function createOffer() {
+  const telegramHandle = removeTelegramURLPrefix(ownerContact.value)
+  console.log('owner_contact: ', telegramHandle)
+  // Encrypt contact to save on the profile when an offer is created
+  const owner_contact_pk = secrets.value.publicKey
```
Comment (samb-local) [inline app/src/ui/components/offers/CreateOffer.vue:79]: Usually, `pk` means private key, can we plz rename this to `owner_contact_pk`?

File: app/src/ui/components/offers/ExpandedOffer.vue [hunk 4]
```diff
@@ -56,13 +72,21 @@ const minMaxCryptoStr = computed(() => {
   return [`${symbol} ${min}`, `${symbol} ${max}`]
 })
 
-function newTrade() {
+async function newTrade() {
+  const telegramHandle = removeTelegramURLPrefix(telegram.value) as string
+  const profile_taker_encrypt_key = secrets.value.publicKey
```
Comment (samb-local) [inline app/src/ui/components/offers/ExpandedOffer.vue:77]: Can we please rename this to something like `profile_taker_encryption_key`, `profile_taker_pub_key` or even `taker_profile_pub_key`?

File: app/tests/arbitration.spec.ts [hunk 5]
```diff
@@ -26,14 +32,29 @@ beforeAll(async () => {
...
+      arbitrators = await adminClient.fetchArbitrators()
+    }
+    expect(arbitrators.filter((arb) => arb.fiat === fiat).length).toBeGreaterThan(0)
+  })
   // Call dispute_escrow on a trade in fiat_deposited state and expects it to be in escrow_disputed state
   it('should have available offers', async () => {
     // Create and fetch offer for trade creation
-    offers[0].denom = { native: process.env.OFFER_DENOM! }
+    const owner_contact = await encryptDataMocked(makerSecrets.publicKey, makerContact)
+    const owner_encrypt_key = makerSecrets.publicKey
```
Comment (samb-local) [inline app/tests/arbitration.spec.ts:52]: Please rename to `owner_encryption_key` or `owner_pub_key`.

File: app/tests/arbitration.spec.ts [hunk 6]
```diff
@@ -75,18 +100,27 @@ describe('arbitration tests', () => {
   })
 
   it('should settle dispute for maker', async () => {
+    const profile_taker_contact = await encryptDataMocked(takerSecrets.publicKey, takerContact)
+    const taker_encrypt_pk = takerSecrets.publicKey
```
Comment (samb-local) [inline app/tests/arbitration.spec.ts:104]: Please rename to `taker_encryption_key`.

File: app/tests/trade.spec.ts [hunk 7]
```diff
@@ -3,35 +3,59 @@ import { TextDecoder, TextEncoder } from 'util'
...
+    }
+    expect(arbitrators.filter((arb) => arb.fiat === fiat).length).toBeGreaterThan(0)
+  })
   // Create Offer
   it('should have an available offer', async () => {
     let myOffers = await makerClient.fetchMyOffers()
     if (myOffers.length === 0) {
-      await makerClient.createOffer(offers[0] as PostOffer)
+      const owner_contact = await encryptDataMocked(makerSecrets.publicKey, makerContact)
+      const owner_encrypt_key = makerSecrets.publicKey
```
Comment (samb-local) [inline app/tests/trade.spec.ts:56]: Please rename to `owner_encryption_key`.

File: app/tests/trade.spec.ts [hunk 8]
```diff
@@ -46,30 +70,38 @@ describe('trade lifecycle happy path', () => {
     })
 
     expect(offersResult.length).toBeGreaterThan(0)
-    const offer = offersResult[0] as GetOffer
+    const offer = offersResult[0]
+    const taker_contact = await encryptDataMocked(offer.owner_encryption_key, takerContact)
     expect(offer).toHaveProperty('id')
-
+    const profile_taker_contact = await encryptDataMocked(takerSecrets.publicKey, takerContact)
+    const profile_taker_encrypt_key = takerSecrets.publicKey
```
Comment (samb-local) [inline app/tests/trade.spec.ts:77]: Please rename to `profile_taker_encryption_key` or `taker_profile_pub_key`.

File: app/tests/trade.spec.ts [hunk 9]
```diff
@@ -94,21 +126,44 @@ describe('trade lifecycle happy path', () => {
...
+    }
+    expect(arbitrators.filter((arb) => arb.fiat === fiat).length).toBeGreaterThan(0)
+  })
   it('should have an available offer', async () => {
     let myOffers = await makerClient.fetchMyOffers()
     if (myOffers.length === 0) {
-      myOffers[0].denom = { native: process.env.OFFER_DENOM! }
-      await makerClient.createOffer(offers[0] as PostOffer)
+      const owner_contact = await encryptDataMocked(makerSecrets.publicKey, makerContact)
+      const owner_encrypt_key = makerSecrets.publicKey
```
Comment (samb-local) [inline app/tests/trade.spec.ts:147]: Please rename to `owner_encryption_key` or `owner_pub_key`.

**Review by samb-local — APPROVED** [review]

**Comment by samb-local** [comment]
Added a few comments regarding variable names. Besides that, LGTM!

**Pull request merged**